crates/ra_ide/src/snapshots/highlight_doctest.html

```diff
@@ -31,40 +31,40 @@
+    <span class="comment documentation">/// </span><span class="keyword">let</span> <span class="keyword">mut</span> <span class="variable declaration mutable">foo</span>: <span class="unresolved_reference">Foo</span> = <span class="unresolved_reference">Foo</span>::<span class="unresolved_reference">new</span>();
+    <span class="comment documentation">/// ```</span>
     <span class="keyword">pub</span> <span class="keyword">const</span> <span class="keyword">fn</span> <span class="function declaration">new</span>() -&gt; <span class="unresolved_reference">Foo</span> {
         <span class="unresolved_reference">Foo</span> { }
     }
 
-    <span class="comment">/// `bar` method on `Foo`.</span>
-    <span class="comment">///</span>
-    <span class="comment">/// # Examples</span>
-    <span class="comment">///</span>
-    <span class="comment">/// ```</span>
-    <span class="comment">/// </span><span class="keyword">let</span> <span class="variable declaration">foo</span> = <span class="unresolved_reference">Foo</span>::<span class="unresolved_reference">new</span>();
-    <span class="comment">///</span>
-    <span class="comment">/// </span><span class="comment">// calls bar on foo</span>
-    <span class="comment">/// </span><span class="macro">assert!</span>(foo.bar());
-    <span class="comment">///</span>
-    <span class="comment">/// </span><span class="comment">/* multi-line
-    </span><span class="comment">/// </span><span class="comment">       comment */</span>
-    <span class="comment">///</span>
-    <span class="comment">/// </span><span class="keyword">let</span> <span class="variable declaration">multi_line_string</span> = <span class="string_literal">"Foo
-    </span><span class="comment">/// </span><span class="string_literal">  bar
-    </span><span class="comment">/// </span><span class="string_literal">         "</span>;
-    <span class="comment">///</span>
-    <span class="comment">/// ```</span>
-    <span class="comment">///</span>
-    <span class="comment">/// ```</span>
-    <span class="comment">/// </span><span class="keyword">let</span> <span class="variable declaration">foobar</span> = <span class="unresolved_reference">Foo</span>::<span class="unresolved_reference">new</span>().<span class="unresolved_reference">bar</span>();
-    <span class="comment">/// ```</span>
+    <span class="comment documentation">/// `bar` method on `Foo`.</span>
+    <span class="comment documentation">///</span>
+    <span class="comment documentation">/// # Examples</span>
+    <span class="comment documentation">///</span>
+    <span class="comment documentation">/// ```</span>
+    <span class="comment documentation">/// </span><span class="keyword">let</span> <span class="variable declaration">foo</span> = <span class="unresolved_reference">Foo</span>::<span class="unresolved_reference">new</span>();
```

Comment:
I still think that code in comments should also be marked up with the `documentation` modifier, or maybe a new modifier like `sample`, `example`, or `usage`, but I'm not sure how the theming controls would play with that, to a degree that it probably won't be useful.  And instead the fallback textmate scopes will provide interesting differentiation (like fontstyle changes other than color).